src/Dierckx.jl

```diff
@@ -952,10 +952,10 @@ function evaluate(spline::Spline2D, x::AbstractVector, y::AbstractVector)
     return z
 end
 
-function evaluate(spline::Spline2D, x::Real, y::Real)
+function evaluate!(wrk::Vector{Float64}, Vspline::Spline2D, x::Real, y::Real)
     ier = Ref{Int32}()
     lwrk = spline.kx + spline.ky + 2
-    wrk = Vector{Float64}(undef, lwrk)
+    @assert length(wrk) == lwrk
```

Comment:
```suggestion
    length(wrk) == lwrk || throw(ArgumentError("Length of work array not equal to required length of `spline.kx + spline.ky + 2 = $(spline.kx + spline.ky + 2)`"))
```

Don't rely on `@assert` for checks that are required for correctness. It may be removed by the compiler, see its docstring.